src/poller.ts

```diff
@@ -0,0 +1,69 @@
+import type { Options } from './types'
+
+export default class Poller {
+  private timeoutId: number
+  private isRunning: boolean
+
+  constructor(
+    private fetchFlagsFn: () => Promise<any>,
+    private configurations: Options,
+    private pollInterval: number,
+    private maxAttempts: number
```

Comment:
This isn't user-configurable anywhere and instead hard-coded in another file. It might make sense to just move it to a constant or default it here and not require it to be passed at all.
```suggestion
    private maxAttempts = 5
```

Reply:
I'll remove it from the constructor as well